src/link/Elf.zig

```diff
@@ -1631,7 +1631,20 @@ fn linkWithLLD(self: *Elf, comp: *Compilation) !void {
         if (self.base.options.each_lib_rpath) {
             var test_path = std.ArrayList(u8).init(self.base.allocator);
             defer test_path.deinit();
+
+            const target_info = try std.zig.system.NativeTargetInfo.detect(self.base.allocator, .{});
```

Comment:
NativeTargetInfo.detect takes a CrossTarget as the second argument. I could probably rebuild it from scratch using what is available, but there should be a better way of doing this that I am not aware of.